openedx_blockstore_relay/tests/test_block_serializer.py

```diff
@@ -0,0 +1,139 @@
+
+from .. import compat
+from ..block_serializer import XBlockSerializer
+from .course_data import TestCourseMixin
+from .xml_test_mixin import XmlTestMixin
+
+
+class XBlockSerializerTestCase(TestCourseMixin, XmlTestMixin, ModuleStoreTestCase):
+    """
+    Tests for XBlockSerializer. Requires a running instance of edX Studio.
+    """
+    maxDiff = None
+
+    def test_sequential(self):
+        """
+        Test that when serializing a sequential/subsection, the children are
+        referenced as <unit>, not <vertical>
+        """
+        block_key = self.course.id.make_usage_key('sequential', 'subsection1_1')
+        result = XBlockSerializer(compat.get_block(block_key))
+        self.assertXmlEqual(result.olx_str, """
+            <sequential display_name="Subsection 1.1">
+                <xblock-include definition="unit/unit1_1_1"/>
+                <xblock-include definition="unit/unit1_1_2"/>
+            </sequential>
+            <!-- Imported from {block_key} using openedx-blockstore-relay -->
+        """.format(block_key=block_key))
+
+    def test_unit(self):
+        """
+        Test serializing a vertical (unit)
+        """
+        block_key = self.course.id.make_usage_key('vertical', 'unit1_1_2')
+        result = XBlockSerializer(compat.get_block(block_key))
+
+        self.assertEqual(result.orig_block_key, block_key)
+        self.assertEqual(result.def_id, "unit/unit1_1_2")
+        self.assertEqual(result.static_files, [])
+        self.assertXmlEqual(result.olx_str, """
+            <unit display_name="Unit 1.1.2">
+                <xblock-include definition="html/html_b"/>
+                <xblock-include definition="video/video_b"/>
+                <xblock-include definition="drag-and-drop-v2/dnd"/>
+            </unit>
+            <!-- Imported from {block_key} using openedx-blockstore-relay -->
+        """.format(block_key=block_key))
+
+    def test_html(self):
+        """
+        Test serializing an HTML block that contains a reference to a
+        contentstore static asset (an asset uploaded via Studio's "Files &
+        Uploads" page)
+        """
+        block_key = self.course.id.make_usage_key('html', 'html_b')
+        result = XBlockSerializer(compat.get_block(block_key))
+
+        self.assertEqual(result.orig_block_key, block_key)
+        self.assertEqual(result.def_id, "html/html_b")
+        self.assertEqual(len(result.static_files), 2)
+        self.assertEqual(result.static_files[0].name, 'html_b.html')
+        self.assertEqual(result.static_files[1].name, 'sample_handout.txt')
+        self.assertEqual(
+            result.static_files[0].data,
+            (
+                '<p>Activate the ωμέγα 13! '
+                '<a href="/static/sample_handout.txt">Instructions.</a></p>'
+            ).encode('utf-8'),
+        )
+        self.assertXmlEqual(result.olx_str, """
+            <html display_name="Unicode and URL test" filename="html_b"/>
```

Comment:
Does the `filename` attribute value not have the suffix?

Reply:
No it does not. That's the existing OLX convention, unchanged.

Reply:
Is this specific to the html block or a runtime/parser feature to append `.html`? Would it make it simpler to treat these the same way as static files?

Reply:
The code that handles that is in the HTML XModule, I believe, not the runtime. However, I haven't looked into it at all.

In the future we could convert it to an XBlock and perhaps give it a new attribute like `src="filename.html"` which will take precedence over `filename="filename"`. Though I think it makes even more sense to just put the HTML inline in the .xml file that defines the HTML block; I'm not really sure what the advantage is of having it in a separate .html file other than perhaps better syntax highlighting and validation in local editor apps that open the file.

Reply:
> The code that handles that is in the HTML XModule, I believe, not the runtime. However, I haven't looked into it at all.

There does seem to be some code in `html_module.py`. I'll take a closer look with converting it to an XBlock.

> Though I think it makes even more sense to just put the HTML inline in the .xml file that defines the HTML block; I'm not really sure what the advantage is of having it in a separate .html file other than perhaps better syntax highlighting and validation in local editor apps that open the file.

We had a conversation around that a few months ago: https://openedx.slack.com/archives/CB29ZP7NJ/p1535471042000100